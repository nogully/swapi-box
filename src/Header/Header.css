.Header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 10px;
  width: 100%;
}

.button-wrapper {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin: 20px;
  justify-content: center;
}

img {
  width: 30%;
}

@media screen and (max-width: 800px){
  img {
    width: 50%;
  }
}

@media screen and (max-width: 400px){
  img {
    width: 60%;
  }
}
