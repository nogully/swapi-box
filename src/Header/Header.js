import React from 'react' 
import logo from './Star_Wars_Yellow_Logo.svg'
import Button from '../Button/Button'
import './Header.css'

const Header = (props) => {
  return (
    <div className="Header"> 
      <img src={logo} className="logo" alt='Star Wars Logo' />
      <div className="button-wrapper">
        <Button buttonType='light'
                buttonText='People'
                function={props.fetchData}/>
        <Button buttonType='light'
                buttonText='Planets'
                function={props.fetchData}/>
        <Button buttonType='light'
                buttonText='Vehicles'
                function={props.fetchData}/>
        <Button buttonType='dark'
                buttonText='Favorites'
                function={props.displayFavorites}/>
      </div>
    </div>
  )
}

export default Header;