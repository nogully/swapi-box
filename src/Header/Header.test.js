import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {
  let wrapper
  let mockFetchData
  let mockDisplayFavorites 
  let mockEvent

  beforeEach(() => {
    mockFetchData = jest.fn()
    mockDisplayFavorites = jest.fn()
    wrapper = shallow(<Header fetchData={ mockFetchData } displayFavorites={ mockDisplayFavorites } />)
  })  

  it('should have a people, planets, vehicles and favorites button', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  })

  it('When a user clicks on People, the fetchData() method is called', () => { 
    wrapper.find('button').first().simulate('click')
    expect(mockFetchData).toHaveBeenCalled()
  })

})
