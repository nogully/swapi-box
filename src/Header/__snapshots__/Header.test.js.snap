// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should have a people, planets, vehicles and favorites button 1`] = `
<div
  className="Header"
>
  <img
    alt="Star Wars Logo"
    className="logo"
    src="Star_Wars_Yellow_Logo.svg"
  />
  <div
    className="button-wrapper"
  >
    <button
      className="light"
      onClick={[Function]}
    >
      people
    </button>
    <button
      className="light"
      onClick={[Function]}
    >
      planets
    </button>
    <button
      className="light"
      onClick={[Function]}
    >
      vehicles
    </button>
    <button
      className="dark"
      onClick={[Function]}
    >
      favorites
    </button>
  </div>
</div>
`;
